src/python/pants/engine/fs.py

```diff
@@ -163,13 +164,32 @@ class DirectoryWithPrefixToStrip(datatype([('directory_digest', Digest), ('prefi
 
 class DirectoryToMaterialize(datatype([('path', str), ('directory_digest', Digest)])):
   """A request to materialize the contents of a directory digest at the provided path."""
-  pass
+
+
+DirectoriesToMaterialize = Collection.of(DirectoryToMaterialize)
+
+
+class MaterializeDirectoryResult(datatype([('output_paths', string_list)])):
+  """Result of materializing a directory, contains the full output paths."""
+
+
+MaterializeDirectoriesResult = Collection.of(MaterializeDirectoryResult)
 
 
 class UrlToFetch(datatype([('url', str), ('digest', Digest)])):
   pass
 
 
+class Workspace:
+  """Abstract handle for operations that touch the real local filesystem."""
+
+  def __init__(self, scheduler):
+    self._scheduler = scheduler
+
+  def materialize_directories(self, directories_to_materialize: Tuple[DirectoryToMaterialize]) -> MaterializeDirectoriesResult:
```

Comment:
Unlike the underlying call, I think that it's ok for this one not to be batch (can always add a batch one next to it later). We started with a batch call primarily because integration with v1 tasks generally needs to write out a _lot_ of stuff... which isn't the case for console rules right now (maybe with `fmt` it will be?)

And given the above, I think that means that the non-batch call could maybe just look like...?
```
def materialize_directory(self, directory_digest: Digest, path_relative_to_buildroot: str = '') -> MaterializeDirectoryResult
```